pkg/tuf/client.go

```diff
@@ -199,9 +199,10 @@ func URLToPath(url string) string {
 	// Strip scheme, replace slashes with dashes
 	// e.g. https://github.github.com/prod-tuf-root -> github.github.com-prod-tuf-root
 	fn := url
-	if len(fn) > 8 && fn[:8] == "https://" {
-		fn = fn[8:]
-	}
+	fn, _ = strings.CutPrefix(fn, "https://")
+	fn, _ = strings.CutPrefix(fn, "http://")
```

Comment:
Does the TUF client require a TLS connection? Is that why stripping off the scheme is OK?

I ask because accessing TUF metadata over HTTP shouldn't be an issue as the metadata is signed (same reason why CA certs/CRLs are served over HTTP sometimes).

Reply:
I think `http://` is used in unit tests (see https://github.com/sigstore/sigstore-go/issues/407).

Reply:
I'm wondering if this needs more changes, that we can't just drop the scheme and need to plumb through whether or not the connection requires TLS to the TUF client.

Reply:
This is just used for the cache directory name, but I can see how we might want to put the scheme in the directory name. The full URL is passed to go-tuf so I think non-TLS is supported there.

Reply:
It’s a niche case to support, maybe this is fine and we just document it? I wouldn’t really expect access to a TUF repo over both http and https and needing multiple distinct caches.

Reply:
TUF itself does not require TLS, the integrity control is managed via the TUF metadata. Only reason for TLS would be to add confidentiality.